necessist/tests/necessist_db_absent/cli_dir_input.toml

```
@@ -0,0 +1,11 @@
+args = ["--no-sqlite", "--dump-candidates", "--framework=rust", "--root=fixtures/cli_dir_input/target_dir"]
```

I would have expected something like this:
```suggestion
args = ["--no-sqlite", "--dump-candidates", "--root=fixtures/cli_dir_input", "src/target_dir"]
```
The `src/target_dir` by itself is what exercises your addition.

So, in other words, `fixtures/cli_dir_input` is the root of a single Cargo project, and `src/target_dir` and `src/adjacent_dir` live inside it.

You might also consider using:
```
args = ["--no-sqlite", "--dump-candidates", "--root=fixtures/cli_dir_input", "src/target_dir", "src/adjacent_dir/some_file.rs"]
```
The idea would be:
- `src/target_dir` exercises the `is_dir` branch in your new code
- `src/adjacent_dir/some_file.rs` exercises the `is_file` branch in your new code

If you do that though, please also have another file in `src/adjacent_dir` (besides `some_file.rs`) whose tests do not get included.

Please tell me if you already know that what I've written won't work.

Also, my "about to travel" comment still applies. :)

---

hey @smoelius thanks for the suggestions, I'll try to implement all of them, albeit it would take some time, let me know when you get back from travelling, in the mean time I'll try to work on this on my own.